test/topology_custom/test_commitlog_segment_data_resurrection.py

```diff
@@ -0,0 +1,124 @@
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+import os
+import logging
+import glob
+import json
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_pinned_cl_segment_doesnt_resurrect_data(manager: ManagerClient):
+    """
+        The tested scenario is as follows:
+        * Two tables, ks1.tbl1 and ks2.tbl2.
+        * A commitlog segment contains writes for both tables.
+        * ks1.tbl1 has very low write traffix, memtables are very rarely
+          flushed, this results in ks1.tbl1 "pinning" any commitlog segment
+          which has writes, that are still in the memtable.
+        * ks2.tbl2 deletes the data, some of which is still in the segment
+          pinned by ks1.tbl1.
+        * ks2.tbl2 flushes the memtable, data gets to sstables, commitlog
+          segments containing writes for both tables are still pinned.
+        * ks2.tbl2 compacts and purges away both the data and the tombstone.
+        * The node has an unclean restart, the pinned segments are replayed and
+          data is resurrected, as the tombstone is already purged.
+
+        This test uses gc_grace_seconds=0 to make the test fast.
+
+        See https://github.com/scylladb/scylladb/issues/14870
+        """
+    server = await manager.server_add(config={
+        "commitlog_sync": "batch",
+        "commitlog_segment_size_in_mb": 1,
+        "enable_cache": False
```

Comment:
Why do you need this? What cache does it disable, the read cache? Why not disable it during the reads with BYPASS CACHE instead of in the configuration?

Reply:
the same like for "commitlog_sync" option: test works without it but maybe author of original scylla test had some reason of using it?

Reply:
The author of the dtest is me. The cache is disabled because it would complicate the test greatly. When flushing the memtable, its content is merged into the cache, so it becomes much harder to control what data exists in memory and what not.